tls/s2n_server_hello.c

```diff
@@ -73,7 +73,7 @@ int s2n_server_hello_recv(struct s2n_connection *conn)
 
     if (conn->server_protocol_version >= S2N_TLS13) {
         /* Check echoed session ID matches */
-        S2N_ERROR_IF(session_id_len != conn->session_id_len || !memcmp(session_id, conn->session_id, session_id_len), S2N_ERR_BAD_MESSAGE);
+        S2N_ERROR_IF(session_id_len != conn->session_id_len || !s2n_constant_time_equals(session_id, conn->session_id, session_id_len), S2N_ERR_BAD_MESSAGE);
```

Comment:
Is there a reference to why we think this needs to be constant time? The session id is public data.

Additionally if we need this to be executed in constant time, the `session_id_len != conn->session_id_len` part of the expression violates that.

Reply:
you've a point, I'll change it back to memcmp